### PR TITLE
Feat/theming

### PR DESCRIPTION
**Pull request by me-jterhorst**
Add a delete Button to the overview list items. 
Add npm dependency to be able to close the menu with a click outside the menu box
Fix headers on sub pages like account, imprint, login
Add darkmode page to allow users to adjust the darkest black for themselves

### CODE REVIEW

File: client/src/pages/Account.js [hunk 1]
```diff
@@ -1,23 +1,21 @@
...
-        </form>
+      <Card isLight={isLight}>
+        <HeaderGoBack />
+        <div className="Card__content">
+          <h2>Account</h2>
+          <form className="Account__form dispFlex">
+            <AccountInput label="Username" value="Username 1" />
+            <AccountInput label="Email" value="JaneWonder@hotmail.de" />
+            <AccountInput label="Password" value="Password" type="password" />
+          </form>
```
Comment (mpagels) [inline client/src/pages/Account.js:17]: The `form` here is not doing anything? No submit. Do you need it?

File: client/src/pages/ListPage.js [hunk 2]
```diff
@@ -3,63 +3,65 @@ import "./ListPage.css";
...
+    })
+    .map((reminder, index) => {
+      return (
+        <section
+          className={`Overview__card dispFlex ${
+            !reminder.trigger.toLowerCase().includes(inputValue.toLowerCase())
+              ? "withHighlight"
+              : ""
+          } `}
+          key={index}
```
Comment (mpagels) [inline client/src/pages/ListPage.js:31]: Using the `index` as a key is an antipattern. You can use it, but why not use something more unique. Maybe you finde something in the `reminder` to use that as a key.

File: client/src/App.js [hunk 3]
```diff
@@ -111,7 +112,24 @@ export default function App() {
...
+    document.documentElement.style.setProperty(
+      "--lightness",
+      `${lightness / 8}%`
+    );
+    localStorage.setItem("lightness", JSON.stringify(lightness));
+  }, [lightness]);
 
+  useEffect(() => {
+    setDarkestblack();
+  }, [setDarkestblack, lightness]);
```
Comment (mpagels) [inline client/src/App.js:132]: No quite sure what happens here. Maybe you can explain it to me?

**Review by mpagels — APPROVED** [review]
LGTM!

Found a lonely form that is doing nothing.  Pls review.

File: client/src/pages/ListPage.js (hunk 2 above)
Comment (me-jterhorst) [reply client/src/pages/ListPage.js:31]: * Wink Wink*

**Review by johnxgaehner — COMMENTED** [review]
Hey Jakob, works smooth!
the darkmode is very small scaled & had problems to see what's happening first. maybe you could turn it into some kind of contrast and darken the white when you lighten the black which could also help tired eyes?

File: client/src/App.js (hunk 3 above)
Comment (me-jterhorst) [reply client/src/App.js:132]: I build my colors with HSL and defined the L part as a variable --lightness.

By default -lightness will be 0% which results in a true black. 
I created my other color values with the same pattern, and just added lightness increments of 15% steps for each lighter color.


Now in my code I can tweak the :root --lightness variable to change (line 123 - 126).
So if I change my base lightness from 0 to 40, 

my
 --color-black-100 will become 5% lighter, 
while my 
--color-black-40 will become 5% + 60% lighter. 



By default, my app should load in true black, unless somebody made a conscious choice to go with a lighter variant. Therefore I opted to write the preffered setting into localstorage and update the lightness setting as soon as the page mounts or something is being adjusted.

**Pull request merged**